ftplugin/crystal.vim

```diff
@@ -59,7 +59,7 @@ nnoremap <Plug>(crystal-format) :<C-u>CrystalFormat<CR>
 
 augroup plugin-ft-crystal
     autocmd!
-    autocmd BufWritePre * if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
+    autocmd BufWritePre <buffer> if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
```

Comment:
If making this auto command buffer-local, checking `&ft` is redundant, I think.  Although it is edge case where user changes its filetype after loading the file as crystal code.